src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,119 @@
+                book.setId(generatedKeys.getObject(1, Long.class));
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add new book: " + book, e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement =
+                    connection.prepareStatement(FIND_BY_ID_SQL);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+
+            if (!resultSet.next()) {
+                throw new RuntimeException("No book with id: " + id);
+            }
+            Book book = new Book(resultSet.getString(TITLE_COLUMN),
+                    resultSet.getObject(PRICE_COLUMN, BigDecimal.class));
+            book.setId(id);
+            return Optional.of(book);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find book with id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> bookList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement =
+                    connection.prepareStatement(FIND_ALL_SQL);
+            ResultSet resultSet = statement.executeQuery();
+
+            while (resultSet.next()) {
+                Book book = new Book(resultSet.getString(TITLE_COLUMN),
+                        resultSet.getObject(PRICE_COLUMN, BigDecimal.class));
+                book.setId(resultSet.getObject(1, Long.class));
+                bookList.add(book);
+            }
+            return bookList;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find books", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement =
+                    connection.prepareStatement(UPDATE_BOOK_SQL, Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, book.getTitle());
+            statement.setObject(2, book.getPrice());
+            statement.setObject(3, book.getId());
+
+            if (statement.executeUpdate() < 1) {
+                throw new RuntimeException("No book for update in database: " + book);
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book: " + book, e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement =
+                    connection.prepareStatement(DELETE_BOOK_SQL, Statement.RETURN_GENERATED_KEYS);
+            statement.setObject(1, book.getId());
+
+            return statement.executeUpdate() > 0;
```

Comment:
The `deleteById` method should return a boolean value depending on the result of `preparedStatement.executeUpdate()`. You are doing this correctly, but make sure to use `statement.executeUpdate()` instead of `preparedStatement.executeUpdate()` to match your variable naming.